Go: Type labels within a package should not be prefixed by package name

Consider this minimal Dafny example 

```dafny
module myclass {
class MyClass
{
  var num: int;

  constructor (x: int) 
  {
      num := x;
  }
}

method main()
{ 
    var c_obj := new MyClass(2);
}
}
```

When compiling Dafny into Go, the compiler produces a Go package `_0_myclass_Compile` where _within the package_, there are variable declarations such as 

```go
var _13_c__obj *_0_myclass_Compile.MyClass
```

When compiling the Go code using the Go compiler, this declaration results in an error of the form

```
src/0_myclass_Compile_/0_myclass_Compile_.go:113:19: undefined: _0_myclass_Compile
```

Removing the package prefix from the declaration, i.e. changing it to 
```go
var _13_c__obj *MyClass
``` 
resolves the issue.

--- a/Test/git-issues/git-issue-588.dfy
+++ b/Test/git-issues/git-issue-588.dfy
@@ -1,0 +1,21 @@
+// RUN: %dafny /compile:0 "%s" > "%t"
+// RUN: %dafny /noVerify /compile:4 /compileTarget:cs "%s" >> "%t"
+// RUN: %dafny /noVerify /compile:4 /compileTarget:js "%s" >> "%t"
+// RUN: %dafny /noVerify /compile:4 /compileTarget:go "%s" >> "%t"
+// RUN: %dafny /noVerify /compile:4 /compileTarget:java "%s" >> "%t"
+// RUN: %diff "%s.expect" "%t"
+
+module myclass {
+  class MyClass {
+    var num: int;
+
+    constructor (x: int) {
+      num := x;
+    }
+  }
+
+  method Main() { 
+    var c_obj := new MyClass(2);
+  }
+}
+

--- a/Test/git-issues/git-issue-588.dfy.expect
+++ b/Test/git-issues/git-issue-588.dfy.expect
@@ -1,0 +1,10 @@
+
+Dafny program verifier finished with 1 verified, 0 errors
+
+Dafny program verifier finished with 0 verified, 0 errors
+
+Dafny program verifier finished with 0 verified, 0 errors
+
+Dafny program verifier finished with 0 verified, 0 errors
+
+Dafny program verifier finished with 0 verified, 0 errors

